### PR TITLE
feat(sampling): Add SDK Client Rate change alert - [TET-244]

### PR DESCRIPTION
**Pull request by priscilawebdev**
**What this PR does:**

- Display an alert, if we think the customer hasn’t updated their SDK sample rate to the recommended yet (if current and recommended SDK sample rates are far apart - at least 50%) 
- Rename `SamplingSDKAlert` component to `SamplingSDKUpgradesAlert`
- Add tests for the new alert
- Add a new amplitude key 

**Preview:**

image

### CODE REVIEW

File: static/app/views/settings/project/server-side-sampling/samplingSDKClientRateChangeAlert.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    return null;
+  }
+
+  return (
+    <Alert
+      type="info"
+      showIcon
+      trailingItems={
+        <Button
+          href={SERVER_SIDE_SAMPLING_DOC_LINK} // TODO(sampling): check if we should use other link here
```
Comment (matejminar) [inline static/app/views/settings/project/server-side-sampling/samplingSDKClientRateChangeAlert.tsx:51]:
```suggestion
          href={`${SERVER_SIDE_SAMPLING_DOC_LINK}product/data-management-settings/server-side-sampling/getting-started/#4-increase-your-sdk-transaction-sample-rate`}
```

File: static/app/views/settings/project/server-side-sampling/utils/index.tsx [hunk 2]
```diff
@@ -51,3 +54,36 @@ export function percentageToRate(rate: number | undefined, decimalPlaces: number
...
+      ? maxSafeSampleRate
+      : undefined;
+
+  const recommendedPercentage = rateToPercentage(recommended);
+  const currentPercentage = rateToPercentage(current);
+
+  const diff =
+    defined(recommendedPercentage) && defined(currentPercentage)
+      ? Math.abs(recommendedPercentage - currentPercentage)
+      : undefined;
```
Comment (matejminar) [inline static/app/views/settings/project/server-side-sampling/utils/index.tsx:82]: We should calculate relative diff, not absolute.

File: static/app/views/settings/project/server-side-sampling/serverSideSampling.tsx [hunk 3]
```diff
@@ -425,14 +426,22 @@ export function ServerSideSampling({project}: Props) {
...
+          <SamplingSDKUpgradesAlert
             organization={organization}
             projectId={project.id}
             rules={rules}
             recommendedSdkUpgrades={recommendedSdkUpgrades}
             onReadDocs={handleReadDocs}
           />
         )}
+        {!!rules.length && (
+          <SamplingSDKClientRateChangeAlert
```
Comment (matejminar) [inline static/app/views/settings/project/server-side-sampling/serverSideSampling.tsx:438]: We probably don't want to show this alert if we show `SamplingSDKUpgradesAlert`.

File: static/app/views/settings/project/server-side-sampling/samplingSDKClientRateChangeAlert.tsx (hunk 1 above)
Comment (priscilawebdev) [reply static/app/views/settings/project/server-side-sampling/samplingSDKClientRateChangeAlert.tsx]: thanks!

File: static/app/views/settings/project/server-side-sampling/utils/index.tsx [hunk 4]
```diff
@@ -51,3 +54,38 @@ export function percentageToRate(rate: number | undefined, decimalPlaces: number
...
+
+  const recommended =
+    defined(maxSafeSampleRate) && !isNaN(maxSafeSampleRate)
+      ? maxSafeSampleRate
+      : undefined;
+
+  let diff: number | undefined = undefined;
+
+  if (defined(recommended) && defined(current)) {
```
Comment (priscilawebdev) [inline static/app/views/settings/project/server-side-sampling/utils/index.tsx:78]: @matejminar is this what you mean?

**Review by andriisoldatenko — APPROVED** [review]

**Pull request merged**